libdnf/common/sack/match_string.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/common/sack/match_string.hpp"
+
+#include <fnmatch.h>
+
+#include <regex>
+#include <stdexcept>
+
+
+namespace libdnf::sack {
+
+
+inline std::string tolower(const std::string & s) {
+    std::string result = s;
+    std::for_each(result.begin(), result.end(), [](char & c) { c = static_cast<char>(::tolower(c)); });
+    return result;
+}
+
+
+bool match_string(const std::string & value, QueryCmp cmp, const std::string & pattern) {
+    bool result = false;
+    switch (cmp) {
+        case QueryCmp::EXACT:
+            result = value == pattern;
+            break;
+        case QueryCmp::NEQ:
+            result = value != pattern;
+            break;
+        case QueryCmp::IEXACT:
+            result = tolower(value) == tolower(pattern);
+            break;
+        case QueryCmp::GLOB:
+            result = fnmatch(pattern.c_str(), value.c_str(), FNM_EXTMATCH) == 0;
+            break;
+        case QueryCmp::IGLOB:
+            result = fnmatch(pattern.c_str(), value.c_str(), FNM_CASEFOLD | FNM_EXTMATCH) == 0;
+            break;
+        case QueryCmp::REGEX: {
+            std::regex re(pattern);
+            result = std::regex_match(value, re);
+        } break;
```

Comment:
Multiple formats is used here. Please could you unify it? Intead of break return can be used, but I don't know what is preferred.

Reply:
Yes I will unify it.